man/man5/zfs-module-parameters.5

```diff
@@ -1688,6 +1674,101 @@ Default value: \fB2\fR.
 .sp
 .ne 2
 .na
+\fBzfs_top_maxinflight\fR (int)
+.ad
+.RS 12n
+Max concurrent I/Os per top-level vdev (mirrors or raidz arrays) allowed during
+legacy scrub or resilver operations.
+.sp
+Default value: \fB32\fR.
+.RE
+
+.sp
+.ne 2
+.na
+\fBzfs_scan_md_maxinflight\fR (int)
+.ad
+.RS 12n
+Max concurrent I/Os allowed during scrub or resilver metadata scanning phase.
+.sp
+Default value: \fB1024\fR.
+.RE
+
+.sp
+.ne 2
+.na
+\fBzfs_scan_leaf_maxinflight\fR (int)
+.ad
+.RS 12n
+Max concurrent I/Os per leaf vdev allowed during scrub or resilver data
+verification phase.
+.sp
+Default value: \fB50\fR.
+.RE
+
+.sp
+.ne 2
+.na
+\fBzfs_scan_issue_strategy\fR (int)
+.ad
+.RS 12n
+Strategy used to determine how data verification should be done during scrubs.
+If set to 1, I/Os will always be issued sequentially. This may help performance
+if the pool's data is very disperate. If set to 2, I/Os will be always be issued
+in size order. This may allow time for gaps between smaller IOs to be filled in
+with other I/Os later. If set to 0, I/Os will be issued in size order, except
+during checkpointing when they will be issued sequentially.
+.sp
+Default value: \fB0\fR.
+.RE
+
+.sp
+.ne 2
+.na
+\fBzfs_scan_max_ext_gap\fR (int)
+.ad
+.RS 12n
+Indicates the largest gap in bytes between scrub / resilver I/Os that will still
+be considered sequential for sorting purposes. Changing this value will not
+affect scrubs or resilvers that are already in progress.
+.sp
+Default value: \fB1048576 (1 MB)\fR.
+.RE
+
+.sp
+.ne 2
+.na
+\fBzfs_scan_fill_weight\fR (int)
+.ad
+.RS 12n
+This tunable affects how scrub and resilver I/O segments are ordered. A higher
+number indicates that we care more about how filled in a segment is, while a
+lower number indicates we care more about the size of the extent without
+considering the gaps within a segment. This value is only tunable upon module
+insertion. Changing the value afterwards will have no affect on scrub
+performance.
+.sp
+Default value: \fB3\fR.
+.RE
+
+.sp
+.ne 2
+.na
+\fBzfs_scan_direct\fR (int)
+.ad
+.RS 12n
+A value of 1 indicates that scrubs and resilvers will be gathered metadata in
+memory before issuing sequential I/O. A value of 0 indicates that the legacy
+algorithm will be used where I/O will be initiated as soon as it is discovered.
+Changing this value to 0 will not affect scrubs or resilvers that are allready
```

Comment:
i just pushed to get another buildbot run going, but i fixed this so it will be in the next push. I'm still not sure what is causing a couple of the failures.....